src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -140,6 +140,8 @@ public CredentialPromptWebRequestHandler(IWebProxy proxy)
                                 var credentials = await HttpHandlerResourceV3
                                     .PromptForProxyCredentials(request.RequestUri, Proxy, cancellationToken);
 
+                                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Consider multiple parallel requests hitting 401 (407 in this case). Then the first one acquired lock successfully faced credential acquisition cancelled exception. Do we want others to prompt user again when they enter this code block after long wait? I say testing cancellation token won't harm there.

It'd be easy to simulate this scenario. Have a fiddler to setup an authenticated proxy. Run massive restore with multiple sources. Cancel authentication request. See if it keeps annoying you with prompts.